### PR TITLE
[green] Added method to get lat/long of a station

### PR DESCRIPTION
**Pull request by zackshapiro**
- Added `lat_and_long` method
- Associated tests for new method
- Fixed one `empty_stations` test to work with the sample data, moved one test to pending, unsure how to test it
- Bumped gemspec version to 0.0.65

### CODE REVIEW

**Comment by zackshapiro** [comment]
@gabceb Wanna take a look at this one, +1 it if it looks good

File: spec/bikeshare_spec.rb [hunk 1]
```diff
@@ -160,15 +160,32 @@
...
     it "should return an array of stations if one or more are offline" do
-      response = BikeShare.any_instance.stub(:offline_stations).and_return([{ :id => 2, :statusKey => 0}, { :id => 8, :statusKey => 0}])
-      response.is_a? Array
-      response.should_not be_nil
+      @response.offline_stations.should_not be_nil
+    end
+  end
+
+  describe "#lat_and_long" do
+    it "should raise an error with a bad station id" do
```
Comment (gabceb) [inline spec/bikeshare_spec.rb]: This test case is already covered by https://github.com/zackshapiro/bikeshare/blob/master/spec/bikeshare_spec.rb#L28

File: spec/bikeshare_spec.rb [hunk 2]
```diff
@@ -160,15 +160,32 @@
...
+      expect { @response.percent_available(0) }.to raise_error
     end
+
+    it "should raise an error with no station id" do
+      expect { @response.percent_available }.to raise_error
+    end
```
Comment (gabceb) [inline spec/bikeshare_spec.rb]: I think this and the previous test case should not be here. Maybe some copy && paste gone bad? :smirk:

**Comment by gabceb** [comment]
I did not find the code associated with your commit message: "Fixed one empty_stations test to work with the sample data, moved one test to pending, unsure how to test it"

**Comment by zackshapiro** [comment]
@gabceb Fixed those things you mentioned, thanks. Sample data has two stations, 38 which is offline and 41 which has zero bikes (it's empty). Made those changes in the test JSON file.

The pending test is at line 63 of `bikeshare_spec.rb`

**Comment by gabceb** [comment]
This is what I see on line 63 when I pull your branch:

```
@response.stations("San Francisco").count.should == 34
```

**Comment by gabceb** [comment]
One last thing, is usually good practice to bump versions on a separate commit so is easier for others to look if necessary. Versions bumps are also increased numerically according to the [SemVer spec](http://semver.org/)

**Comment by zackshapiro** [comment]
Noted. Thanks. Not sure why you're not seeing appropriate changes. Try pulling again?

**Comment by gabceb** [comment]
Have you pushed to this branch on Github? If you go to the "Files Changed" section of this PR I should be able to see your changes but still cannot see them.

**Comment by zackshapiro** [comment]
Yeah, I have. I see the changes in my Files Changed section. Not sure why the discrepancy

**Comment by gabceb** [comment]
Maybe you meant line 163? -> https://github.com/zackshapiro/bikeshare/pull/8/files#diff-ce1786ba1c822bdf89a438a3ec0a2528R163

**Comment by zackshapiro** [comment]
Yep, my mistake

**Comment by gabceb** [comment]
Ha. I was looking for an actual pending test like the one below. That way it will show on the tests results. Let me take a look at it see if I can help.

``` ruby
describe "an example" do
  it "is implemented but waiting" do
    pending("something else getting finished")
    this_should_not_get_executed
  end
end
```

**Comment by zackshapiro** [comment]
Hah whoops, my bad. Thanks man

**Comment by gabceb** [comment]
The solution I propose is to add a url parameter to the initialize method to get this test to work. If you think this is a good idea you can pull my lat-long branch.

Here is my commit with the changes https://github.com/gabceb/bikeshare-1/commit/3b9518c6c3722afa0c2b0afd155cbb199b5f3fbf

This commit also make the json responses into separate files to keep the code clean.

**Comment by zackshapiro** [comment]
@gabceb For the state of this gem right now and the needs that it feels, that seems excessive to alter the init function for the purposes of running tests alone. I'd rather merge this branch, put these changes on a shelf and revisit it in the future.

**Comment by gabceb** [comment]
Allow me to disagree with you on this one @zackshapiro.

If you look at the commit code you will notice that the url passed is only used if available, otherwise the default url will be used so this won't break previous versions of the gem. If you think about it, the url is not added **only** for the test but as an option for developers to change that url if needed. The url parameter also happens to be used by one of the tests (actually 2 tests because I added a test for this change as well) :smirk:

Anyways, I do agree that this gem as it is does what it is supposed to do so don't feel bad if you don't want to pull my changes but I do think my changes are an improvement to the gem.

**Comment by zackshapiro** [comment]
@gabceb After sleeping on it, I really like this. Was your branch branched off of my branch? I'd like to merge yours into master

**Comment by gabceb** [comment]
Yeah. Just pull my branch lat-long from git@github.com:gabceb/bikeshare-1.git

**Comment by zackshapiro** [comment]
Merged. Good work sir

**Pull request merged**